subsys/net/lib/lwm2m/lwm2m_rd_client.c

```diff
@@ -920,6 +920,13 @@ static int sm_registration_done(void)
 		client.trigger_update = false;
 		client.update_objects = false;
 
+		/*
+		 * Update tx time directly here to prevent invalid timestamp
+		 * when performing check should ENGINE_REGISTRATION_DONE_RX_OFF
+		 * be generated below.
+		 */
+		engine_update_tx_time();
```

Comment:
I wonder, maybe it'd be better to move `engine_update_tx_time()` into `lwm2m_send_message_async()` instead of covering corner cases in lwm2m_rd_client? I am aware that it's not the actual timestamp of putting the packet on the wire, but the delay between queueing the packet and the actual send call should be negligible, given the timeout value (which is 93 seconds by default). What do you think?

Reply:
> but the delay between queueing the packet and the actual send call should be negligible,

Sounds good. Initially I did not want to put it there (`lwm2m_send_message_async`), but in the function where the bug actually occurs so fix is more obvious. But since the delay really is negligible I'll move it there.